src/main/java/bio/terra/service/dataset/DatasetRequestValidator.java

```diff
@@ -363,6 +367,32 @@ private void validateSchema(DatasetSpecificationModel schema, Errors errors) {
         }
     }
 
+    private void validateRegion(DatasetRequestModel datasetRequest, Errors errors) {
+        if (datasetRequest.getRegion() != null) {
+            if (datasetRequest.getCloudPlatform() == null) {
+                errors.rejectValue("region", "InvalidRegionForPlatform",
+                    "Cannot set a region when a cloudPlatform is not provided.");
+            } else {
+                boolean supported = false;
+                switch (datasetRequest.getCloudPlatform()) {
```

Comment:
Probably not worth doing now, but if we have a lot of cases where we switch on platform type, we may want to isolate all the per-platform decisions into an interface and construct it using a factory.  This would help isolate the per-platform logic from the code that uses it. Normally I'd say we should put the logic or factory code in the enum itself but since it's a generated class we can't do that here.